src/pages/index.js

```diff
@@ -1,38 +1,86 @@
 import "./index.css";
 
 //import all the classes
-import { initialCards, selectors, config } from "../utils/constants";
+import { selectors, config } from "../utils/constants";
 import Card from "../components/Card";
 import FormValidator from "../components/FormValidator";
 import Section from "../components/Section";
 import PopupWithImage from "../components/PopupWithImage";
 import PopupWithForm from "../components/PopupWithForm";
 import UserInfo from "../components/UserInfo";
-
+import Api from "../components/Api";
+import PopupWithConfirm from "../components/PopupWithConfirm";
 //Create instances of the classes
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d78649ed-fd14-41f7-9a2b-04c3fb13cc28",
+    "content-type": "application/json",
+  },
+});
+
+const cardSection = new Section(
+  {
+    renderer: (data) => {
+      cardSection.addItems(createCard(data));
+    },
+  },
+  selectors.cardSection
+);
+
 const createCard = (data) => {
-  const card = new Card(data, "#card-template", () => {
-    cardPreviewPopup.open(data);
-  });
+  const card = new Card(
+    data,
+    "#card-template",
+    () => {
+      cardPreviewPopup.open(data);
+    },
+    function handleCardDelete(card) {
+      confirmDeletePopup.open();
+      confirmDeletePopup.setSubmitAction(() => {
+        confirmDeletePopup.setLoading(true, "Deleting");
+        api
+          .removeCard(card.getId())
+          .then(() => {
+            card._handleTrashIcon();
+            confirmDeletePopup.close();
+          })
+          .catch((err) => {
+            console.log(err);
+          })
+          .finally(() => {
+            confirmDeletePopup.setLoading(false, "Deleting");
+      });
+    },
+    (card) => {
+      const id = card.getId();
+      if (card.isLiked()) {
+        console.log("disliking");
+        api.dislikeCard(id)
```

Comment:
* When I like/dislike cards, then reload the page, the hearts show the initial state (so, it doesn't change on the server) . Please, correct my remark from here https://github.com/UffyLane/se_project_aroundtheus/pull/1#discussion_r2072416281